src/App/Fossa/Report.hs

```diff
@@ -83,48 +85,9 @@ reportMain baseUri apiKey logSeverity timeoutSeconds reportType override = do
   hPutStrLn stderr "Timed out while waiting for build/issues scan"
   exitFailure
 
-waitForBuild
-  :: (Has Diagnostics sig m, MonadIO m, Has Logger sig m)
-  => URI
-  -> ApiKey -- ^ api key
-  -> ProjectRevision
-  -> m ()
-waitForBuild baseUri key revision = do
-  build <- Fossa.getLatestBuild baseUri key revision
-  case Fossa.buildTaskStatus (Fossa.buildTask build) of
-    Fossa.StatusSucceeded -> pure ()
-    Fossa.StatusFailed -> fatal BuildFailed
-    otherStatus -> do
-      logSticky $ "[ Waiting for build completion... last status: " <> viaShow otherStatus <> " ]"
-      liftIO $ threadDelay (pollDelaySeconds * 1000000)
-      waitForBuild baseUri key revision
-
-waitForIssues
-  :: (Has Diagnostics sig m, MonadIO m, Has Logger sig m)
-  => URI
-  -> ApiKey -- ^ api key
-  -> ProjectRevision
-  -> m ()
-waitForIssues baseUri key revision = do
-  issues <- Fossa.getIssues baseUri key revision
-  case Fossa.issuesStatus issues of
-    "WAITING" -> do
-      liftIO $ threadDelay (pollDelaySeconds * 1000000)
-      waitForIssues baseUri key revision
-    _ -> pure ()
-
-pollDelaySeconds :: Int
-pollDelaySeconds = 8
-
 timeout
   :: Int -- ^ number of seconds before timeout
   -> IO a
   -> IO (Maybe a)
-timeout seconds act = either id id <$> Async.race (Just <$> act) (threadDelay (seconds * 1000000) *> pure Nothing)
-
-data WaitError
-  = BuildFailed -- ^ we encountered the FAILED status on a build
-  deriving (Show, Generic)
-
-instance ToDiagnostic WaitError where
-  renderDiagnostic BuildFailed = "The build failed. Check the FOSSA webapp for more details."
+-- timeout seconds act = either id id <$> Async.race (Just <$> act) (threadDelay (seconds * 1_000_000) *> pure Nothing)
+timeout seconds act = either id id <$> Async.race (Just <$> act) (threadDelay (seconds * 1_000_000) $> Nothing)
```

Comment:
great call on `-XNumericUnderscores`. and I was questioning whether to use `$>` here when initially writing this; happy to see it here now.